tensorflow/core/kernels/data/range_dataset_op_test.cc

```diff
@@ -0,0 +1,207 @@
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/framework/dataset.h"
+#include "tensorflow/core/framework/fake_input.h"
+#include "tensorflow/core/framework/node_def_builder.h"
+#include "tensorflow/core/framework/partial_tensor_shape.h"
+#include "tensorflow/core/framework/variant.h"
+#include "tensorflow/core/kernels/data/iterator_ops.h"
+#include "tensorflow/core/kernels/ops_testutil.h"
+#include "tensorflow/core/platform/test.h"
+
+namespace tensorflow {
+namespace data {
+namespace {
+
+class RangeDatasetOpTest : public OpsTestBase {
+ public:
+  Status InitOp() { return InitOpWithGraphVersion(TF_GRAPH_DEF_VERSION); }
+
+  // Only use this directly if you have a deprecated op that you need to test.
+  Status InitOpWithGraphVersion(int graph_def_version) {
+    OpKernel* kernel = nullptr;
+    Status status = CreateOpKernel(device_type_, device_.get(), allocator(),
+                                   flr_, node_def_, graph_def_version, &kernel);
+    kernel_.reset(kernel);
+    if (kernel_ != nullptr) input_types_ = kernel_->input_types();
+    return status;
+  }
+
+  Status InitThreadPool(int thread_num = 1) {
+    CHECK_GE(thread_num, 1);
+    thread_pool_ = new thread::ThreadPool(Env::Default(), ThreadOptions(),
+                                          "inter_op", thread_num);
+    return Status::OK();
+  }
+
+  Status InitFunctionLibraryRuntime(const std::vector<FunctionDef>& flib,
+                                    thread::ThreadPool* thread_pool = nullptr,
+                                    int cpu_num = 2) {
+    SessionOptions options;
+    auto* device_count = options.config.mutable_device_count();
+    device_count->insert({"CPU", cpu_num});
+    std::vector<std::unique_ptr<Device>> devices;
+    TF_CHECK_OK(DeviceFactory::AddDevices(
+        options, "/job:localhost/replica:0/task:0", &devices));
+    device_mgr_ = absl::make_unique<DeviceMgr>(std::move(devices));
+
+    FunctionDefLibrary proto;
+    for (const auto& fdef : flib) *(proto.add_function()) = fdef;
+    lib_def_.reset(new FunctionLibraryDefinition(OpRegistry::Global(), proto));
+    fdef_lib_ = lib_def_->ToProto();
+
+    OptimizerOptions opts;
+    pflr_.reset(new ProcessFunctionLibraryRuntime(
+        device_mgr_.get(), Env::Default(), TF_GRAPH_DEF_VERSION, lib_def_.get(),
+        opts, thread_pool, nullptr /* cluster_flr */));
+
+    flr_ = pflr_->GetFLR("/job:localhost/replica:0/task:0/cpu:0");
+
+    if (thread_pool == nullptr) {
+      runner_ = [](std::function<void()> fn) { fn(); };
+    } else {
+      runner_ = [thread_pool](std::function<void()> fn) {
+        thread_pool->Schedule(std::move(fn));
+      };
+    }
+
+    return Status::OK();
+  }
+
+  Status GetDatasetOutputFromContext(int output_index) {
+    auto* tensor = GetOutput(output_index);
+    TF_CHECK_OK(GetDatasetFromVariantTensor(*tensor, &dataset_));
```

Comment:
s/TF_CHECK_OK/TF_RETURN_ERROR_IF/

also, I think you will need to call `dataset_->Ref()` here and add a destructor that calls `dataset_->Unref()`